### ISSUE TITLE
Inference with predict_on_batch returns prediction with low score for a certain class

### ISSUE THREAD

**Issue by mdatres** [fizyr/keras-retinanet#1526]
Dear all, 

I'm facing a strange behavior in retinanet 1.0. I have recently updated retinanet from version 0.5 to version 1.0. I have two classes, one with a validation mAP of 0.87 and the other with a validation mAP of 0.3 both in version 0.5 and 1.0. I have trained retinanet 0.5 on the training set, and, after doing that, I have converted the trained model to an inference model. I have used predict_on_batch (as in examples) to get the predictions on the test set and I have saved them in a csv file. The predictions obtained with version 0.5 contain both classes with a score that goes from 0.2 to 1, as expected. But the predictions obtained using version 1.0 provide the predictions of one class (the one with 0.87 mAP) with a score between 0 and 1, while for the prediction of the other class the score is always below 0.5. I set up the score threshold to 0.5 during training in both cases (version 0.5 and 1.0). I don't understand why in version 1.0 the model seems not to find any object of the second class with a score greater than 0.5. Can someone help me? 

Best regards, 
mdatres
